beacon-chain/core/blocks/payload_test.go

```diff
@@ -409,6 +413,106 @@ func Test_ExecutionEnabled(t *testing.T) {
 	}
 }
 
+func Test_ValidatePayloadWhenMergeCompletes(t *testing.T) {
+	tests := []struct {
+		name    string
+		payload *ethpb.ExecutionPayload
+		header  *ethpb.ExecutionPayloadHeader
+		err     error
+	}{
+		{
+			name:    "merge incomplete",
+			payload: emptyPayload(),
+			header:  emptyPayloadHeader(),
+			err:     nil,
+		},
+		{
+			name:    "validate passes",
+			payload: emptyPayload(),
+			header: func() *ethpb.ExecutionPayloadHeader {
+				h := emptyPayloadHeader()
+				h.ParentHash = bytesutil.PadTo([]byte{'a'}, fieldparams.RootLength)
```

Comment:
Empty payload and payload header with `a` parent hash are equal?

Reply:
Ah. The test setup is bogus. `ParentHash` is not used. It was just comparing zeros so it passed